Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -125,44 +132,53 @@ public void clearCache() {
     if (cancellationToken != null && cancellationToken.isCancelled()) {
       return Task.cancelled();
     }
-    final File file = getCacheFile(state);
+    final File cacheFile = getCacheFile(state);
     return Task.call(new Callable<Boolean>() {
       @Override
       public Boolean call() throws Exception {
-        return file.exists();
+        return cacheFile.exists();
       }
     }, Task.BACKGROUND_EXECUTOR).continueWithTask(new Continuation<Boolean, Task<File>>() {
       @Override
       public Task<File> then(Task<Boolean> task) throws Exception {
         boolean result = task.getResult();
         if (result) {
-          return Task.forResult(file);
+          return Task.forResult(cacheFile);
         }
         if (cancellationToken != null && cancellationToken.isCancelled()) {
           return Task.cancelled();
         }
 
+        // Generate the temp file path for caching ParseFile content based on ParseFile's url
+        // The reason we do not write to the cacheFile directly is because there is no way we can
+        // verify if a cacheFile is complete or not. If download is interrupted in the middle, next
+        // time when we download the ParseFile, since cacheFile has already existed, we will return
+        // this incomplete cacheFile
```

Comment:
This makes sense since although we use a `TaskQueue` in `ParseFile`, there could potentially be multiple instances of the same `ParseFile` without LDS.

Would it make sense to instead keep a mapping of URL -> `Task<File>` so that we return the `Task<File>` if the same URL is attempted to be fetched multiple times? This would reduce the amount of requests as well as remove the potential error when we update the file pointer when a developer is reading it due to concurrent fetches on the same `ParseFile`.